### PR TITLE
quic: remove experimental status and add it to the default config

### PR DESCRIPTION
**Pull request by RubenKelevra**

### CODE REVIEW

File: core/node/groups.go [hunk 1]
```diff
@@ -130,7 +130,6 @@ func LibP2P(bcfg *BuildCfg, cfg *config.Config) fx.Option {
 		maybeProvide(libp2p.BandwidthCounter, !cfg.Swarm.DisableBandwidthMetrics),
 		maybeProvide(libp2p.NatPortMap, !cfg.Swarm.DisableNatPortMap),
 		maybeProvide(libp2p.AutoRelay, cfg.Swarm.EnableAutoRelay),
-		maybeProvide(libp2p.QUIC, cfg.Experimental.QUIC),
```
Comment (Stebalien) [inline core/node/groups.go:133]:
```go
 		fx.Provide(libp2p.QUIC),
```

Comment (Stebalien) [reply core/node/groups.go:133]: (nvm)

**Comment by Stebalien** [comment]
You'll also need to change DefaultTransports in core/node/libp2p/transport.go to `simpleOpt(libp2p.ChainOptions(libp2p.DefaultTransports, libpp2quic.NewTransport))`.

**Comment by RubenKelevra** [comment]
@Stebalien The private network test is expected to fail since QUIC doesn't support private networks. I guess we have to modify this test, to remove QUIC from it.

**Comment by Stebalien** [comment]
@RubenKelevra hm, I forgot about that. I think we need to avoid constructing the QUIC transport when a swarm key is present.

**Comment by RubenKelevra** [comment]
Or we just turn QUIc only on, if we're starting to listening on a QUIC port.

This way the user can determine if he/she wants QUIC with the listening definition.

Since the test only listens to TCP it wouldn't fail in this case.


Feel free to push the commit here, if you decided on that. Both is a bit over my head right now :)

**Pull request merged**